nexus/api.go

```diff
@@ -176,3 +196,53 @@ func addContextTimeoutToHTTPHeader(ctx context.Context, httpHeader http.Header)
 	httpHeader.Set(HeaderRequestTimeout, time.Until(deadline).String())
 	return httpHeader
 }
+
+type Link struct {
+	// URL encoded information about the link
+	// It cannot contain the following chars: ',', '';', '='.
+	URL string
```

Comment:
We need to manually validate anyway since we allow workflow ID to contain any char. For the URL, the workflow ID will have to be percent-encoded. Otherwise, decoding the Link header will be a pain.

Reply:
Not following this exactly. Whatever you do with workflow ID extraction from the URL is unrelated to this general-purpose, Temporal-agnostic library. I was just suggesting that data type because it is guaranteed a valid URL if in that data type. But if we want validated string instead that's fine.